frontend/src/app/key-result-dialog/key-result-dialog.component.ts

```diff
@@ -113,4 +164,49 @@ export class KeyResultDialogComponent implements OnInit {
     const errors = this.keyResultForm.get(name)?.errors;
     return errors == null ? [] : Object.keys(errors);
   }
+
+  saveEmittedData(keyResult: KeyResultEmitDTO) {
+    this.isMetricKeyResult = keyResult.keyresultType == 'metric';
+    if (keyResult.keyresultType == 'metric') {
+      let metricKeyResult = keyResult as KeyResultEmitMetricDTO;
+      this.keyResultForm.patchValue({
+        baseline: metricKeyResult.baseline,
+        stretchGoal: metricKeyResult.stretchGoal,
+        unit: metricKeyResult.unit,
+      });
+    } else {
+      let ordinalKeyResult = keyResult as KeyResultEmitOrdinalDTO;
+      this.keyResultForm.patchValue({
+        commitZone: ordinalKeyResult.commitZone,
+        targetZone: ordinalKeyResult.targetZone,
+        stretchZone: ordinalKeyResult.stretchZone,
+      });
+    }
+  }
+
+  filter(value: string): Observable<User[]> {
+    const filterValue = value.toLowerCase();
+    return this.users$.pipe(
+      map((users) =>
+        users.filter(
+          (user) =>
+            user.firstname.toLowerCase().includes(filterValue) ||
+            user.lastname.toLowerCase().includes(filterValue) ||
+            user.username.toLowerCase().includes(filterValue),
+        ),
+      ),
+    );
+  }
+
+  getUserNameById(user: User): string {
+    if (user === null || user === undefined) {
+      return '';
+    }
+
+    return user.firstname + ' ' + user.lastname;
+  }
+
+  validOwner() {
```

Comment:
Remove this method

Reply:
Could not be removed. Is still used to print the error messages.